link/kprobe_multi.go

```diff
@@ -138,7 +138,7 @@ func (kml *kprobeMultiLink) Unpin() error {
 	return fmt.Errorf("unpin kprobe_multi: %w", ErrNotSupported)
 }
 
-var haveBPFLinkKprobeMulti = internal.FeatureTest("bpf_link_kprobe_multi", "5.18", func() error {
+var haveBPFLinkKprobeMulti = internal.FeatureTest("bpf_link_kprobe_multi", "0.0", func() error {
```

Comment:
I think this version number should stay, otherwise our CI will miss this test failing.

Reply:
If we don't remove the version number, tests fail for me with:

```
Feature 'bpf_link_kprobe_multi' isn't supported even though kernel v5.19.11 is newer than v5.18
```

Maybe we can add some magic (`-X` define or a build tag) so the version number checks happen in CI tests, but not locally?

Reply:
Hmm, I'd like to keep the default (= when running locally) to throw an error when this happens. Otherwise developing a new feature test becomes cumbersome.

How about an environment variable that contains separated feature test names for which to ignore an error result? Basically, in https://github.com/cilium/ebpf/blob/8fceee5ca41b47a59739a14e2f6877886dc89023/internal/testutils/feature.go#L58-L62 check `tb.Name()` against an env var and call `tb.Skip()`. Brownie points if we only split the env var once or so.

This means you have to opt in to ignore the result, but you can stick that variable in your `.bashrc` or something. If your kernel changes and the test automagically isn't skipped anymore.

Reply:
I also think the version number should stay, Build tags are a hard nack from me, and ldflags are quite cumbersome to specify.

We've also recently adopted envs in `cilium/cilium` to gate test behaviours. All in all, I think it's the lesser evil, because they can be set ambiently and are easy to slap onto any command, as there might be arbitrary tools/scripts in between your shell and the actual test binaries reading the env.

I'd probably go with a single env here that basically trusts the verdict of our `haveBPFXYZ()` functions and doesn't execute the kernel version check. We could add this kind of skip in `testutils.CheckFeatureTest()` or `testutils.checkKernelVersion()`, although the latter might cause more unintended side effects.

As long as the default doesn't change and we don't relax CI.